en: simplify gke deployment doc (#701)
pingcap/docs-tidb-operator: en/deploy-on-gcp-gke.md

<!-- markdownlint-disable MD029 -->
<!-- markdownlint-disable MD037 -->

This document describes how to deploy a TiDB cluster on GCP Google Kubernetes Engine (GKE).

## Prerequisites

> Before deploying a TiDB cluster on AWS EKS, make sure the following requirements are satisfied:

Reviewer: ```suggestion
Before deploying a TiDB cluster on GCP GKE, make sure the following requirements are satisfied:
```

* Install [Helm](https://helm.sh/docs/intro/install/): used for deploying TiDB Operator.
* Install [gcloud](https://cloud.google.com/sdk/gcloud): a command-line tool used for creating and managing GCP services.
* Complete the operations in the **Before you begin** section of [GKE Quickstart](https://cloud.google.com/kubernetes-engine/docs/quickstart#before-you-begin).

    This guide includes the following contents:

    * Enable Kubernetes APIs
    * Configure enough quota

## Deploy the cluster

### Configure the GCP service

Configure your GCP project and default region:

{{< copyable "shell-regular" >}}

```bash
gcloud config set core/project <gcp-project>
gcloud config set compute/region <gcp-region>
```

### Create a GKE cluster

1. Create a GKE cluster and a default node pool:

    {{< copyable "shell-regular" >}}

    ```shell
    gcloud container clusters create tidb --machine-type n1-standard-4 --num-nodes=1
    ```

>     * The command above creates a regional cluster. In the regional cluster, the node will create three zones in the region to ensure high availability.

Reviewer: ```suggestion
    * The command above creates a regional cluster.
```

>     * The `--num-nodes=1` option indicates that one node is created in each zone, so there are three nodes in total.

Reviewer: ```suggestion
    * The `--num-nodes=1` option indicates that one node is created in each zone, so there are three nodes in total if there are three zones in the region to ensure high availability.
```

The LoadBalancer created for your TiDB cluster is an intranet LoadBalancer. You can create a [bastion host](https://cloud.google.com/solutions/connecting-securely#bastion) in the cluster VPC to access the database.

{{< copyable "shell-regular" >}}

```shell
gcloud compute instances create bastion \
  --machine-type=n1-standard-4 \
  --image-project=centos-cloud \
  --image-family=centos-7 \
  --zone=<your-region>-a
```

> **Note:**
>
> `<your-region>-a` is the `a` zone in the region of the cluster, such as `us-central1-a`. You can also create the bastion host in other zones in the same region.

### Install the MySQL client and connect

After the bastion host is created, you can connect to the bastion host via SSH and access the TiDB cluster via the MySQL client.

1. Connect to the bastion host via SSH:

    {{< copyable "shell-regular" >}}

    ```shell
    gcloud compute ssh tidb@bastion
    ```

2. Install the MySQL client:

    {{< copyable "shell-regular" >}}

    ```shell
    sudo yum install mysql -y
    ```

3. Connect the client to the TiDB cluster:

    {{< copyable "shell-regular" >}}

    ```shell
    mysql -h <tidb-nlb-dnsname> -P 4000 -u root
    ```

>     `<tidb-nlb-dnsname>` is the LoadBalancer IP of the TiDB service. You can view the IP in the `EXTERNAL-IP` field by executing `kubectl get svc basic-tidb -n tidb-cluster`.

Reviewer 2: ```suggestion
    `<tidb-nlb-dnsname>` is the LoadBalancer IP of the TiDB service. You can view the IP in the `EXTERNAL-IP` field of the `kubectl get svc basic-tidb -n tidb-cluster` execution result.
```

    Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

    Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

    MySQL [(none)]> show status;
    +--------------------+--------------------------------------+
    | Variable_name      | Value                                |
    +--------------------+--------------------------------------+
    | Ssl_cipher         |                                      |
    | Ssl_cipher_list    |                                      |
    | Ssl_verify_mode    | 0                                    |
    | Ssl_version        |                                      |
    | ddl_schema_version | 22                                   |
    | server_id          | 717420dc-0eeb-4d4a-951d-0d393aff295a |
    +--------------------+--------------------------------------+
    6 rows in set (0.01 sec)
    ```

> **Note:**
>
> By default, TiDB (starting from v4.0.2) periodically shares usage details with PingCAP to help understand how to improve the product. For details about what is shared and how to disable the sharing, see [Telemetry](https://docs.pingcap.com/tidb/stable/telemetry).

> ## Monitor the cluster

Reviewer 2: This is different from the Chinese `Grafana 监控`. The following titles might be verbs or verb phrases such as `Scale out`, `Upgrade`.
```suggestion
## Monitor
```

> Obtain the LoadBalancer domain name of Grafana:

Reviewer: ```suggestion
Obtain the LoadBalancer IP of Grafana:
```

{{< copyable "shell-regular" >}}

```shell
kubectl -n tidb-cluster get svc basic-grafana
```

For example:

```
$ kubectl -n tidb-cluster get svc basic-grafana
NAME                     TYPE           CLUSTER-IP      EXTERNAL-IP      PORT(S)               AGE
basic-grafana            LoadBalancer   10.15.255.169   34.123.168.114   3000:30657/TCP        35m
```

> In the output above, the `EXTERNAL-IP` column is the LoadBalancer domain name.

Reviewer: ```suggestion
In the output above, the `EXTERNAL-IP` column is the LoadBalancer IP.
```

> You can access the `<grafana-lb>:3000` address using your web browser to view monitoring metrics. Replace `<grafana-lb>` with the LoadBalancer domain name.

Reviewer: ```suggestion
You can access the `<grafana-lb>:3000` address using your web browser to view monitoring metrics. Replace `<grafana-lb>` with the LoadBalancer IP.
```

The initial Grafana login credentials are:

- User: admin
- Password: admin

## Upgrade

To upgrade the TiDB cluster, edit the `spec.version` by executing `kubectl edit tc basic -n tidb-cluster`.

The upgrade process does not finish immediately. You can watch the upgrade progress by executing `kubectl get pods -n tidb-cluster --watch`.

## Scale out

Before scaling out the cluster, you need to scale out the corresponding node pool so that the new instances have enough resources for operation.

> The following example shows how to scale out the `tikv` node pool of the `tidb` cluster to 6 nodes:

Reviewer 2: ```suggestion
The following example shows how to scale out the `tikv` node pool of the `tidb` cluster to have 6 nodes:
```

{{< copyable "shell-regular" >}}

```shell
gcloud container clusters resize tidb --node-pool tikv --num-nodes 2
```

> **Note:**
>

> > In the regional cluster, the nodes are created in 3 zones. Therefore, after scaling out, the number of nodes is 2 * 3 = 6.

Reviewer 2: ```suggestion
> In the regional cluster, the nodes are created in 3 zones. Therefore, after scaling out, the number of nodes is `2 * 3 = 6`.
```

> After that, execute `kubectl edit tc basic -n tidb-cluster`, and modify each component's `replicas` to the desired number of replicas. The scaling-out process is then completed.

Reviewer 2: ```suggestion
After that, execute `kubectl edit tc basic -n tidb-cluster` and modify each component's `replicas` to the desired number of replicas. The scaling-out process is then completed.
```

## Deploy TiFlash and TiCDC

### Create new node pools

* Create a node pool for TiFlash:

    {{< copyable "shell-regular" >}}

    ```shell
    gcloud container node-pools create tiflash --cluster tidb --machine-type n1-highmem-8 --num-nodes=1 \
        --node-labels dedicated=tiflash --node-taints dedicated=tiflash:NoSchedule
    ```

* Create a node pool for TiCDC:

    {{< copyable "shell-regular" >}}

    ```shell
    gcloud container node-pools create ticdc --cluster tidb --machine-type n1-standard-4 --num-nodes=1 \
        --node-labels dedicated=ticdc --node-taints dedicated=ticdc:NoSchedule
    ```

### Configure and deploy

* To deploy TiFlash, configure `spec.tiflash` in `tidb-cluster.yaml`. For example:

    ```yaml
    spec:
      ...
      tiflash:
        baseImage: pingcap/tiflash
        replicas: 1
        storageClaims:
        - resources:
            requests:
            storage: 100Gi
        nodeSelector:
          dedicated: tiflash
        tolerations:
        - effect: NoSchedule
          key: dedicated
          operator: Equal
          value: tiflash
    ```

    > **Warning:**
    >

>     > TiDB Operator automatically mount PVs **in the order of the configuration** in the `storageClaims` list. Therefore, if you need to add disks for TiFlash, make sure that you add the disks **only to the end of the original configuration** in the list. In addition, you must **not** alter the order of the original configuration.

Reviewer 2: ```suggestion
    > TiDB Operator automatically mounts PVs **in the order of the configuration** in the `storageClaims` list. Therefore, if you need to add disks for TiFlash, make sure that you add the disks **only to the end of the original configuration** in the list. In addition, you must **not** alter the order of the original configuration.
```
